Backpack-SwiftUI/TextArea/Classes/BPKTextArea.swift

```diff
@@ -0,0 +1,131 @@
+    
+    private enum BorderConstants {
+        static let cornerRadius: BPKCornerRadius = .xs
+        static let lineWidth: CGFloat = 1
+    }
+    
+    @Environment(\.colorScheme) var colorScheme
+    @Binding private var value: String
+    private let charLimit: Int
+    private let placeholder: String
+    private var onValueChange: (String) -> Void = { _ in }
+    
+    /// Creates a `BPKTextArea`.
+    ///
+    /// - Parameters:
+    ///   - text: The text to display in the text field.
+    ///   - placeholder: The placeholder text to display when the text field is empty.
+    ///   - charLimit: Limit of character allowed to enter
+    ///   - onValueChange: Callback
+    public init(
+        _ value: Binding<String>,
+        placeholder: String = "",
+        charLimit: Int,
+        onValueChange: @escaping (String) -> Void = { _ in }
+    ) {
+        self._value = value
+        self.placeholder = placeholder
+        self.charLimit = charLimit
+        self.onValueChange = onValueChange
+    }
+    
+    var threeLineHeight: CGFloat {
+        let font = BPKFontStyle.bodyDefault
+        return font.lineHeight * 3
+    }
+    
+    @ViewBuilder
+    var textEditorContent: some View {
+        if #available(iOS 16.0, *) {
+            TextEditor(text: $value)
+                .font(style: .bodyDefault)
+                .scrollContentBackground(.hidden)
+        } else {
+            TextEditor(text: $value)
+                .font(style: .bodyDefault)
+        }
+    }
+    
+    public var body: some View {
+        ZStack(alignment: .topLeading) {
+            // Text editor
+            textEditorContent
+                .frame(height: threeLineHeight)
+                .foregroundColor(.textPrimaryColor)
+                .padding(.vertical, TextEditorConstants.verticalPadding)
+                .padding(.horizontal, TextEditorConstants.horizontalPadding)
+                .onChange(of: value) { newValue in
+                    if newValue.count > charLimit {
+                        onValueChange(String(newValue.prefix(charLimit)))
+                    } else {
+                        onValueChange(newValue)
+                    }
+                }
+            
+            // Placeholder
+            BPKText(placeholder, style: .bodyDefault)
+                .foregroundColor(.textSecondaryColor)
+                .allowsHitTesting(false)
+                .padding(.vertical, PlaceholderConstants.verticalPadding)
+                .padding(.horizontal, PlaceholderConstants.horizontalPadding)
+                .opacity(value.isEmpty ? 1 : 0)
+            
+            // Border
+            RoundedRectangle(cornerRadius: BorderConstants.cornerRadius)
+                .strokeBorder(.lineColor, lineWidth: BorderConstants.lineWidth)
+        }
+        .background(.surfaceDefaultColor)
+        .clipShape(
+            RoundedRectangle(cornerRadius: BorderConstants.cornerRadius)
+        )
+        .accessibilityLabel(placeholder)
```

Comment:
This might break when the text area has content, can you verify the label is correct and reads the actual contents when there are contents?

Reply:
Changed to dynamic label.